### PR TITLE
*: separate use of kv.Storage and tikv.Storage

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
We are working on moving out the logic in `tikv.Storage` that is not directly related to TiKV. In the future `tikv.Storage` will no longer completely cover `kv.Storage`. Therefore, we need to distinguish where `Storage` is used, that is, decide which interface to use according to the method that needs to be used.
Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- Move `Name` and `Describe` methods out of `tikv.Storage`.
- Use `kv.Storage` or `tikv.Storage` separatly.

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by andylokandy — APPROVED** [review]

File: store/mockstore/unistore.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ type mockStorage struct {
 	*tikv.KVStore
 }
 
+// NewMockStorage wraps tikv.KVStore as kv.Storage.
+func NewMockStorage(tikvStore *tikv.KVStore) kv.Storage {
```
Comment (andylokandy) [inline store/mockstore/unistore.go:49]: Why is it called 'mockstorage' while the tikv.KVStore is a real component?

Comment (disksing) [reply store/mockstore/unistore.go:49]: `KVStore` is real but it connects to an in-memory cluster 😄

**Review by AndreMouche — APPROVED** [review]
/LGTM

**Review by andylokandy — APPROVED** [review]

**Pull request merged**